rust/src/dataset.rs

```diff
@@ -309,7 +309,8 @@ impl Dataset {
                 return Err(Error::from(batch.as_ref().unwrap_err()));
             }
         } else {
-            return Err(Error::EmptyDataset);
```

Comment:
We might be able to remove this `EmptyDataset` as error.

Reply:
Yeah, indeed. It's gone in latest version.